### PR TITLE
Added nostrip to CONFIG to avoid illegal binary strips on WSL

### LINKED ISSUE #1059: Trying to install kiwi-desktop results in error, due to strip
After building kiwix-desktop and all its dependencies, when running `sudo make install`, I get the following error log:

```/usr/lib/qt5/bin/qmake -install qinstall -exe kiwix-desktop /usr/local/bin/kiwix-desktop
strip /usr/local/bin/kiwix-desktop
/usr/lib/qt5/bin/qmake -install qinstall -exe /mnt/c/Users/jim31/OneDrive/Desktop/SGOURDAS/kiwix-desktop/resources/org.kiwix.desktop.desktop /usr/local/share/applications/org.kiwix.desktop.desktop
strip /usr/local/share/applications/org.kiwix.desktop.desktop
strip: /usr/local/share/applications/org.kiwix.desktop.desktop: file format not recognized
make: [Makefile:2545: install_desktop_file] Error 1 (ignored)
/usr/lib/qt5/bin/qmake -install qinstall -exe /mnt/c/Users/jim31/OneDrive/Desktop/SGOURDAS/kiwix-desktop/resources/org.kiwix.desktop.appdata.xml /usr/local/share/metainfo/org.kiwix.desktop.appdata.xml
strip /usr/local/share/metainfo/org.kiwix.desktop.appdata.xml
strip: /usr/local/share/metainfo/org.kiwix.desktop.appdata.xml: file format not recognized
make: [Makefile:2555: install_metainfo_file] Error 1 (ignored)
/usr/lib/qt5/bin/qmake -install qinstall -exe /mnt/c/Users/jim31/OneDrive/Desktop/SGOURDAS/kiwix-desktop/resources/icons/kiwix/24/kiwix-desktop.png /usr/local/share/icons/hicolor/24x24/apps/kiwix-desktop.png
strip /usr/local/share/icons/hicolor/24x24/apps/kiwix-desktop.png
strip: /usr/local/share/icons/hicolor/24x24/apps/kiwix-desktop.png: file format not recognized
make: [Makefile:2565: install_app_icon24] Error 1 (ignored)
/usr/lib/qt5/bin/qmake -install qinstall -exe /mnt/c/Users/jim31/OneDrive/Desktop/SGOURDAS/kiwix-desktop/resources/icons/kiwix/32/kiwix-desktop.png /usr/local/share/icons/hicolor/32x32/apps/kiwix-desktop.png
strip /usr/local/share/icons/hicolor/32x32/apps/kiwix-desktop.png
strip: /usr/local/share/icons/hicolor/32x32/apps/kiwix-desktop.png: file format not recognized
make: [Makefile:2575: install_app_icon32] Error 1 (ignored)
/usr/lib/qt5/bin/qmake -install qinstall -exe /mnt/c/Users/jim31/OneDrive/Desktop/SGOURDAS/kiwix-desktop/resources/icons/kiwix/48/kiwix-desktop.png /usr/local/share/icons/hicolor/48x48/apps/kiwix-desktop.png
strip /usr/local/share/icons/hicolor/48x48/apps/kiwix-desktop.png
strip: /usr/local/share/icons/hicolor/48x48/apps/kiwix-desktop.png: file format not recognized
make: [Makefile:2585: install_app_icon48] Error 1 (ignored)
/usr/lib/qt5/bin/qmake -install qinstall -exe /mnt/c/Users/jim31/OneDrive/Desktop/SGOURDAS/kiwix-desktop/resources/icons/kiwix/128/kiwix-desktop.png /usr/local/share/icons/hicolor/128x128/apps/kiwix-desktop.png
strip /usr/local/share/icons/hicolor/128x128/apps/kiwix-desktop.png
strip: /usr/local/share/icons/hicolor/128x128/apps/kiwix-desktop.png: file format not recognized
make: [Makefile:2595: install_app_icon128] Error 1 (ignored)
/usr/lib/qt5/bin/qmake -install qinstall -exe /mnt/c/Users/jim31/OneDrive/Desktop/SGOURDAS/kiwix-desktop/resources/icons/kiwix/256/kiwix-desktop.png /usr/local/share/icons/hicolor/256x256/apps/kiwix-desktop.png
strip /usr/local/share/icons/hicolor/256x256/apps/kiwix-desktop.png
strip: /usr/local/share/icons/hicolor/256x256/apps/kiwix-desktop.png: file format not recognized
make: [Makefile:2605: install_app_icon256] Error 1 (ignored)
/usr/lib/qt5/bin/qmake -install qinstall -exe /mnt/c/Users/jim31/OneDrive/Desktop/SGOURDAS/kiwix-desktop/resources/icons/kiwix/scalable/kiwix-desktop.svg /usr/local/share/icons/hicolor/scalable/apps/kiwix-desktop.svg
strip /usr/local/share/icons/hicolor/scalable/apps/kiwix-desktop.svg
strip: /usr/local/share/icons/hicolor/scalable/apps/kiwix-desktop.svg: file format not recognized
make: [Makefile:2615: install_app_svg] Error 1 (ignored)
/usr/lib/qt5/bin/qmake -install qinstall -exe /mnt/c/Users/jim31/OneDrive/Desktop/SGOURDAS/kiwix-desktop/resources/org.kiwix.desktop-mime.xml /usr/local/share/mime/packages/org.kiwix.desktop-mime.xml
strip /usr/local/share/mime/packages/org.kiwix.desktop-mime.xml
strip: /usr/local/share/mime/packages/org.kiwix.desktop-mime.xml: file format not recognized
make: [Makefile:2625: install_mime_file] Error 1 (ignored)
```

In the makefile the strip command is being executed on incompatible file types like pngs etc, but AFAIK `strip` should be run on files that contain machine code. So, to continue with the install process I just commented all the lines running that command and everything worked after.

As a fix, should I go ahead and remove the lines running `strip` on incorrect data types, or is there something else that should be done?

OS Version: Ubuntu 20.04.6 LTS, through WSL2
kiwix-desktop Version: 2.3.1

### PR DESCRIPTION
**Pull request by sgourdas**
Fixes #1059

Added the `nostrip` flag to `CONFIG` in the `kiwix-desktop.pro` in case of WSL

### CODE REVIEW

**Comment by kelson42** [comment]
I don't think we want to just remove it everywhere. The best solution would be to do this only for WSL. The second best wozld be to do that via qmake/makefile option and document the case properly.

**Review by kelson42 — CHANGES_REQUESTED** [review]
@sgourdas Have you found a way to detect WSL properly?

**Comment by sgourdas** [comment]
@kelson42 Based on [this](https://superuser.com/questions/1749781/how-can-i-check-if-the-environment-is-wsl-from-a-shell-script) it seems like checking for the existence of `/proc/sys/fs/binfmt_misc/WSLInterop` is the most reliable method, IMO as well, to check for WSL. I added a conditional check for this in the `.pro` file.

If this is fine for you also, I can squash the commits so we can merge.

**Comment by kelson42** [comment]
@sgourdas Thx, I will give a try and verify if it works fine for Linux at least.

**Comment by kelson42** [comment]
@sgourdas This piece of code deserves a small code comment explaining the situation/solution

File: kiwix-desktop.pro [hunk 1]
```diff
@@ -8,6 +8,10 @@ QT       += core gui network
 QT       += webenginewidgets
 QT       += printsupport
 
+#Determine if the host system is WSL based on the existence of "/proc/sys/fs/binfmt_misc/WSLInterop"
+WSL_CHECK = $$system(test -f /proc/sys/fs/binfmt_misc/WSLInterop && echo true || echo false)
+equals(WSL_CHECK, "true"): CONFIG += nostrip
+
```
Comment (mgautierfr) [inline kiwix-desktop.pro:14]: This comment is not so useful. We already see that we are checking existence of `/proc/sys/fs/binfmt_misc/WSLInterop` in the code. With a small variable renaming we can easily give the hint about what we are checking.

If we add comment, it should be why we do this, not what or how we are doing it.

I suggest:

```
# install script fails to detect if file is a binary or not on wsl and then try to strip all
# installed files, which obviously not work for all files. So don't strip on WSL.
ARE_WE_IN_WSL = $$system(test -f /proc/sys/fs/binfmt_misc/WSLInterop && echo true || echo false)
equals(ARE_WE_IN_WSL, "true"): CONFIG += nostrip
```

Comment (kelson42) [reply kiwix-desktop.pro:14]: LGTM, please do and squash everything in one commit so we can merge.

**Review by kelson42 — APPROVED** [review]
LGTM

**Pull request merged**